### ISSUE TITLE
RSA encryption/decryption doesn't restore original value when given message starts with 0

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#103]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/98) by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

This should return `#vu8(0 0 0 1)`

```
#!scheme
(import (rnrs)
	(crypto))

(define keypair (generate-key-pair RSA))

(define msg #vu8(0 0 0 1))

(let ((enc-ci (cipher RSA (keypair-public keypair) :padding #f))
      (dec-ci (cipher RSA (keypair-private keypair) :padding #f)))
  (decrypt dec-ci (encrypt enc-ci msg)))
```
But return s `#vu8(0)`

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Bouncy Castle returns `#vu8(1)` with nopadding. This seems kind of inevitable. All what we can do is that ensuring to return proper value when padding is specified. Thus it should also return `#vu8(1)` if it's not padded.

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Modified not to strip leading 0 when RSA cipher uses padding. (Fixes #98)
